pkgs/development/compilers/idris2/default.nix

```diff
@@ -1,22 +1,33 @@
-{ lib, stdenv, fetchFromGitHub, makeWrapper
-, clang, chez
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, clang
+, chez
+, gmp
+, racket
+, zsh
 }:
 
+# NOTICE: An `idris2WithPackages` is available at: https://github.com/claymager/idris2-pkgs
+
 # Uses scheme to bootstrap the build of idris2
 stdenv.mkDerivation rec {
   pname = "idris2";
-  version = "0.3.0";
+  version = "0.4.0";
 
   src = fetchFromGitHub {
     owner = "idris-lang";
     repo = "Idris2";
     rev = "v${version}";
-    sha256 = "0sa2lpb7n6xqfknwld9rzm4bnb6qcd0ja1n63cnc5v8wdzr8q7kh";
+    sha256 = "105jybjf5s0k6003qzfxchzsfcpsxip180bh3mdmi74d464d0h8g";
   };
 
   strictDeps = true;
-  nativeBuildInputs = [ makeWrapper clang chez ];
-  buildInputs = [ chez ];
+  nativeBuildInputs = [ makeWrapper clang ]
+    ++ lib.optional stdenv.isDarwin [ zsh ];
+  propagatedNativeBuildInputs = [ chez racket ];
```

Comment:
```suggestion
  buildInputs = [ chez ];
  # We do not add any propagatedNativeBuildInputs because
  # we do not want the executables idris2 produces to depend on the nix-store.
  # As such, it is left to the user to guarantee chez and/or racket are available
  # in the path during execution of programs compiled with idris2.
```

Reply:
Changed the message a bit:
```
+  # We do not add any propagatedNativeBuildInputs because we do not want the
+  # executables idris2 produces to depend on the nix-store. As such, it is left
+  # to the user to guarantee chez (or any other codgen dependency) is available
+  # in the path during compilation of programs with idris2.
```